src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowProvider.cs

```diff
@@ -79,7 +80,11 @@ public IVsInteractiveWindow Create(int instanceId)
             vsWindow.SetLanguage(LanguageServiceGuid, evaluator.ContentType);
 
             // the tool window now owns the engine:
-            vsWindow.InteractiveWindow.TextView.Closed += new EventHandler((_, __) => evaluator.Dispose());
+            vsWindow.InteractiveWindow.TextView.Closed += new EventHandler((_, __) => 
```

Comment:
this can potentially leak. you need to unsubscribe from Closed event handler within the OnClose event handler.